### PR TITLE
Improve search performance

### PR DESCRIPTION
**Pull request by pablolazaro**
Hey @Guillaume-Docquier. First I would like to let you know how much I appreciate all the work you've done with the website. Is an incredible tool and is going to help a lot to the Crowfall community. Yesterday I've found the site and one thing that I realized is the text search was a bit slowly (like it had some kind of stuttering due to layout shifts between key strokes). I'm not a very experienced React developer but I tought that maybe I could bring something to the table.

Basically the idea is to reduce component re-renders between key strokes and also try to memoize as much as possible. As far as I know there is no much room to improve regarding `includes` function performance (unless you're thinking in adding some external library) so I left it as is.

Please take this just like a suggestion and let me know if it is a good start point.

PSA: I had to add `@material-ui/core`, application doesn't start without it

### CODE REVIEW

File: src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx [hunk 1]
```diff
@@ -3,39 +3,35 @@ import "./ItemSearch.css";
...
     const [matchedItems, setMatchedItems] = useState(allItems);
     const [selectedItemId, setSelectedItemId] = useState(null);
 
-    useEffect(() => {
-        const timeout = setTimeout(() => {
-            setMatchedItems(allItems.filter(item => {
-                if (filter.length === 0) {
-                    return true;
-                }
+    const filteredItems = useMemo(() => (filter.length === 0 ? allItems : filterItems(filter)), [filter]);
```
Comment (Guillaume-Docquier) [inline src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx:25]: Small styling preference, I'd remove this empty line. Since the two statements are closely related an not referenced anywhere else, I like it better when it's a block.

Same thing goes for the empty line at line 29.

File: src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx [hunk 2]
```diff
@@ -3,39 +3,35 @@ import "./ItemSearch.css";
...
```
Comment (Guillaume-Docquier) [inline src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx:24]: Here the parentheses around the arrow function's body are not required.

File: src/crowcraft/package.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "homepage": "https://guillaume-docquier.github.io/crowcraft/",
   "dependencies": {
+    "@material-ui/core": "^4.11.2",
```
Comment (Guillaume-Docquier) [inline src/crowcraft/package.json:7]: This is very curious. As you'll imagine, I am able to run the application without including `@material-ui/core`.  
Running `npm list -g` to see if I have this package installed globally outputs the following:
```
├── @angular/cli@1.5.0
├── create-react-app@4.0.3
├── electron@1.7.9
├── eslint@7.14.0
├── foundation-cli@2.2.6
├── nodemon@2.0.7
├── npm-check-updates@8.1.1
├── npm@7.5.1
├── ts-node@5.0.0
├── typeorm@0.1.12
└── typescript@2.7.2
```

I'll investigate to see if one of these packages includes `@material-ui/core`, but could you elaborate on what error you get when trying to start the project without this dependency? I'd prefer not including dependencies we do not need (if we do not need them)

**Review by Guillaume-Docquier — COMMENTED** [review]
First of all, thank you for contributing! The search feels a lot nicer now that it is smooth!
This is overall very good work. Before merging I just want to dig into the `@material-ui/core` issue, to which I've left a comment

File: src/crowcraft/package.json (hunk 3 above)
Comment (pablolazaro) [reply src/crowcraft/package.json]: My bad. I was using an old version of Node.js and NPM. With the latest version it works like a charm. Reverted here 
c409ea6

File: src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx (hunk 2 above)
Comment (pablolazaro) [reply src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx]: Removed here cafa3cf!

File: src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx (hunk 1 above)
Comment (pablolazaro) [reply src/crowcraft/src/components/crafting/item-search/ItemSearch.jsx:24]: I hope I understand you properly. I've made the changes here 9695885. Let me know if there is anything else 👍

**Review by Guillaume-Docquier — APPROVED** [review]
lgtm!
